addons/mail/static/src/views/form/form_compiler.js

```diff
@@ -37,18 +36,19 @@ function compileChatter(node, params) {
                 break;
         }
     }
-    const chatterContainerXml = createElement("Chatter");
+    const chatterContainerXml = createElement("t");
     setAttributes(chatterContainerXml, {
+        "t-component": "__comp__.mailComponents.Chatter",
         hasActivities,
         hasFollowers,
         hasMessageList,
         hasParentReloadOnAttachmentsChanged,
         hasParentReloadOnFollowersUpdate,
         hasParentReloadOnMessagePosted,
         isAttachmentBoxVisibleInitially,
-        threadId: params.threadId,
-        threadModel: params.threadModel,
-        webRecord: params.webRecord,
+        threadId: "__comp__.props.record.model.root.resId or undefined",
+        threadModel: "__comp__.props.record.model.root.resModel",
+        webRecord: "__comp__.props.record.model.root",
```

Comment:
@rfr-odoo `__comp__.props.record.model.root` is the main form view, `__comp__.props.record` is the record for which we're rendering a form view. I think `__comp__.props.record.resId`, `__comp__.props.record.resModel` and `__comp__.props.record` would be more appropriate. This explains the problem you had earlier with the chatter not being linked to the correct model. That being said, we also need a mechanism to prevent having chatter in x2many though (it worked before as I was handled by the controller, which we don't have in an x2many).